src/sdk/glacier.js

```diff
@@ -136,31 +152,50 @@ class GlacierBackend {
         // the migration.
         if (stat.blocks === 0) return false;
 
-        const restore_status = GlacierBackend.get_restore_status(stat.xattr, new Date(), file);
+        const restore_status = Glacier.get_restore_status(stat.xattr, new Date(), file);
         if (!restore_status) return false;
 
-        return restore_status.state === GlacierBackend.RESTORE_STATUS_CAN_RESTORE;
+        return restore_status.state === Glacier.RESTORE_STATUS_CAN_RESTORE;
     }
 
     /**
      * get_restore_status returns status of the object at the given
      * file_path
-     * 
+     *
      * NOTE: Returns undefined if `user.storage_class` attribute is not
      * `GLACIER`
-     * @param {nb.NativeFSXattr} xattr 
-     * @param {Date} now 
-     * @param {string} file_path 
+     * @param {nb.NativeFSXattr} xattr
+     * @param {Date} now
+     * @param {string} file_path
      * @returns {nb.RestoreStatus | undefined}
      */
     static get_restore_status(xattr, now, file_path) {
-        if (xattr[GlacierBackend.STORAGE_CLASS_XATTR] !== s3_utils.STORAGE_CLASS_GLACIER) return;
+        const storage_class = Glacier.storage_class_from_xattr(xattr);
+        if (storage_class !== s3_utils.STORAGE_CLASS_GLACIER) {
+            return;
+        }
+
+        if (Glacier.is_externally_managed(xattr)) {
```

Comment:
if xattr.storage_class!=GLACIER we don't really want to consider it as externally managed and skip this case.